plans/run.pp

```diff
@@ -1,8 +1,12 @@
 # Starts a Puppet agent run on the specified targets.
 # Note: This plan may cause issues when run in Puppet Enterprise.
 # @param targets The targets to start a Puppet agent run on.
+# @param noop if true, all runs will use --noop
+# @param environment the desired puppet code environment
 plan puppet_agent::run (
-  TargetSpec $targets
+  TargetSpec $targets,
+  Boolean $noop = false,
```

Comment:
I considered it. When you currently set $noop to false (or keep the default), the parameter won't be passed at all to puppet agent. So it's not a breaking change. I think that's a sane default.

Reply:
Sorry for the late response. My point was that when `noop=true` in `puppet.conf`, this module plan are currently unusable to actually do changes because puppet run in noop mode all the time because of the configuration setting.

With this PR, we have a chance to "fix" it by adding an explicit `--no-noop` parameter to the puppet command when we set `noop=false`.

I heard some people are setting `noop=true` in puppet.conf and manually run the agent in no-noop mode when they detect drift. These users are currently not able to do this using the provided tasks and plans.

Reply:
Of course, in this case the 'sane default' is not to change noop unless a value was passed.

In any case, this PR as it is today is very fine for me if you think the value is not worth it.